src/Features/CSharp/Portable/Completion/KeywordRecommenders/ElseKeywordRecommender.cs

```diff
@@ -23,35 +23,21 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
 
             var token = context.TargetToken;
 
-            var statement = token.GetAncestor<StatementSyntax>();
-            var ifStatement = statement.GetAncestorOrThis<IfStatementSyntax>();
-
-            if (statement == null || ifStatement == null)
+            // We have to consider all ancestor if statements of the last token until we find a match for this 'else':
+            // while (true)
+            //     if (true)
+            //         while (true)
+            //             if (true)
+            //                 Console.WriteLine();
+            //             else
+            //                 Console.WriteLine();
+            //     $$
+            foreach (var ifStatement in token.GetAncestors<IfStatementSyntax>())
             {
-                return false;
-            }
-
-            // cases:
-            //   if (goo)
-            //     Console.WriteLine();
-            //   |
-            //   if (goo)
-            //     Console.WriteLine();
-            //   e|
-            if (token.IsKind(SyntaxKind.SemicolonToken) && ifStatement.Statement.GetLastToken(includeSkipped: true) == token)
-            {
-                return true;
-            }
-
-            // if (goo) {
-            //     Console.WriteLine();
-            //   } |
-            //   if (goo) {
-            //     Console.WriteLine();
-            //   } e|
-            if (token.IsKind(SyntaxKind.CloseBraceToken) && ifStatement.Statement is BlockSyntax && token == ((BlockSyntax)ifStatement.Statement).CloseBraceToken)
-            {
-                return true;
+                if (ifStatement.Statement.GetLastToken(includeZeroWidth: true) == token)
```

Comment:
warrants a comment.   but not a big deal.

Reply:
fixed in 0c495ca42df1962d6406ee762f351ed6c9824186